src/dmd/typesem.d

```diff
@@ -3079,8 +3073,16 @@ void resolve(Type mt, const ref Loc loc, Scope* sc, out Expression pe, out Type
 
     void visitMixin(TypeMixin mt)
     {
-        auto o = mt.compileTypeMixin(loc, sc);
+        RootObject o = mt.obj;
+        if (o)
+        {
+            pe = o.isExpression();
+            pt = o.isType();
+            ps = o.isDsymbol();
+            return;
+        }
```

Comment:
Comment on why are you checking this.

Reply:
Remember: `RootObject obj; // cached result of semantic analysis.`

I'm checking if the object is already cached, if so just return it in the corresponding variable.